packages/sdk/src/client.ts

```diff
@@ -565,84 +586,90 @@ export class Client
         const userEvents = [
             await makeEvent(
                 this.signerContext,
                 make_UserPayload_Inception({
                     streamId: streamIdAsBytes(userStreamId),
+                    appAddress: metadata?.appAddress ?? bin_fromHexString(''),
                 }),
             ),
         ]
+        const encoded = metadata ? stripUndefinedMetadata(metadata) : metadata
         const response = await this.rpcClient.createStream({
             events: userEvents,
             streamId: streamIdAsBytes(userStreamId),
-            metadata: metadata,
+            metadata: encoded,
         })
         return unpackStream(response.stream, this.opts?.unpackEnvelopeOpts)
     }
 
     private async createUserMetadataStream(
         userMetadataStreamId: string | Uint8Array,
-        metadata: { spaceId: Uint8Array } | undefined,
+        metadata?: { spaceId?: Uint8Array; appAddress?: Uint8Array },
     ): Promise<ParsedStreamResponse> {
         const userDeviceKeyEvents = [
             await makeEvent(
                 this.signerContext,
                 make_UserMetadataPayload_Inception({
                     streamId: streamIdAsBytes(userMetadataStreamId),
+                    appAddress: metadata?.appAddress ?? bin_fromHexString(''),
                 }),
             ),
         ]
-
+        const encoded = metadata ? stripUndefinedMetadata(metadata) : metadata
         const response = await this.rpcClient.createStream({
             events: userDeviceKeyEvents,
             streamId: streamIdAsBytes(userMetadataStreamId),
-            metadata: metadata,
+            metadata: encoded,
         })
         return unpackStream(response.stream, this.opts?.unpackEnvelopeOpts)
     }
 
     private async createUserInboxStream(
         userInboxStreamId: string | Uint8Array,
-        metadata: { spaceId: Uint8Array } | undefined,
+        metadata?: { spaceId?: Uint8Array; appAddress?: Uint8Array },
     ): Promise<ParsedStreamResponse> {
         const userInboxEvents = [
             await makeEvent(
                 this.signerContext,
                 make_UserInboxPayload_Inception({
                     streamId: streamIdAsBytes(userInboxStreamId),
+                    appAddress: metadata?.appAddress ?? bin_fromHexString(''),
                 }),
             ),
         ]
-
+        const encoded = metadata ? stripUndefinedMetadata(metadata) : metadata
         const response = await this.rpcClient.createStream({
             events: userInboxEvents,
             streamId: streamIdAsBytes(userInboxStreamId),
-            metadata: metadata,
+            metadata: encoded,
         })
         return unpackStream(response.stream, this.opts?.unpackEnvelopeOpts)
     }
 
     private async createUserSettingsStream(
         inUserSettingsStreamId: string | Uint8Array,
-        metadata: { spaceId: Uint8Array } | undefined,
+        metadata?: { spaceId?: Uint8Array; appAddress?: Uint8Array },
```

Comment:
these signatures should be `createUserXStream(streamId, appId, metadata)`
the metadata is passed along with the request, the app id is part of the inception, lets not mix them